.openshift-ci/sensor_versions.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+"""
+Returns the latest patches of the last 4 major versions
+"""
+
+import sys
+import re
+import subprocess
+
+def isReleaseTagGitCLI(version):
+    return bool(re.search(r"^\d+\.\d+\.\d+$", version))
+
+def filterGitCLITags(rawtags):
+    filteredtags = []
+    for t in rawtags:
+        if isReleaseTagGitCLI(t):
+            filteredtags.append(t)
+    return set(filteredtags)
+
+def cliOutputToTags(stdoutput):
+    separated = stdoutput.decode(encoding="utf-8").splitlines()
+    return separated
+
+def transformTagsToNumbers(tags):
+    numTags = []
+    for t in tags:
+        numTags.append(splitVersion(t))
+    return sorted(numTags)
+
+def splitVersion(version):
+    digits = re.search(r"(\d+)\.(\d+)\.\D*(\d+)", version)
+    return int(digits.group(1)), int(digits.group(2)), int(digits.group(3))
+
+def getLatestTags(tags, num_versions):
```

Comment:
I think this can be simplified. You don't need to convert it to numbers and then back to strings. `sorted` will work just fine in numbered strings. 

Here's what I suggest you do:
1. Make a dictionary all versions based on their major versions: `{ 3.69: [3.69.0, 3.69.1 ...], 3.70: [ ... ], ...`
2. Sort the dictionary keys and get the top 4
3. Sort each bucket and get top 1

Here's a drop-in replacement that should work the same way without needing `transformTagsToNumbers`:

```python3
def getLatestTags(tags, num_versions):
    top_patch_version = defaultdict(int)
    for t in tags:
        [major, minor, patch] = t.split('.')
        k = '.'.join([major, minor])
        top_patch_version[k] = max(top_patch_version[k], int(patch))
    top_major_versions = sorted(list(top_patch_version.keys()), reverse=True)[:num_versions]
    return [t + '.' + str(top_patch_version[t]) for t in top_major_versions]
```

I quickly tested with your script and it _seems_ to work. But please, if you use the function above make sure it works for all the cases you thought of. 

This is just a suggestion to simplify the code.

Reply:
Good idea with the dicts. Just tested it out and it works, but only if the MAIN_IMAGE_TAG is the latest, eg. this will always grab the 4 latest major sensor versions but it should grab the version corresponding to central and the previous 3. I'll adapt it accordingly.